server/src/main/java/org/elasticsearch/action/bulk/IncrementalBulkService.java

```diff
@@ -14,74 +14,72 @@
 import org.elasticsearch.action.support.ActiveShardCount;
 import org.elasticsearch.client.internal.Client;
 import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.Releasable;
 import org.elasticsearch.core.Releasables;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.index.IndexingPressure;
-import org.elasticsearch.rest.action.document.RestBulkAction;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Supplier;
 
+import static org.elasticsearch.common.settings.Setting.boolSetting;
+
 public class IncrementalBulkService {
 
+    public static final Setting<Boolean> INCREMENTAL_BULK = boolSetting(
+        "rest.incremental_bulk",
+        true,
```

Comment:
I'd probably want to flip this before merge to main? Unless we are confident in having the feature enabled without a rollout plan. Can be a follow-up.

Reply:
I will add a ticket and make it a blocker for merge into main.